### ISSUE TITLE
strange behavior when using docker-compose restart xxx

### ISSUE THREAD

**Issue by titangate** [docker/compose#5239]
Docker on Mac
docker-compose version 1.16.1, build 6d1ac21
docker-py version: 2.5.1
CPython version: 2.7.12
OpenSSL version: OpenSSL 1.0.2j  26 Sep 2016

docker-compose.yml:

```version: '3'
services:
  wtf:
    image: "ubuntu:16.04"
    volumes:
      - .:/var
    working_dir: /var
    entrypoint: /bin/sh /var/starter.sh
  wtf2:
    image: "ubuntu:16.04"
    volumes:
      - .:/var
    working_dir: /var
    entrypoint: sleep infinity
```

starter.sh

```
echo "heyheyhey"

sleep infinity
```

now in terminal A I ran `docker-compose up`,
on the terminal B i ran `docker-compose restart wtf` repeatedly.
here's output from terminal A
<img width="264" alt="screen shot 2017-10-05 at 8 22 32 pm" src="https://user-images.githubusercontent.com/655011/31258498-8921d4b2-aa0d-11e7-876a-dc5745be5659.png">

each color change indicates a restart
every time the the container was restarted, the script was run more than once.

What's even stranger is that if i want to guard the other scripts from running, i tried to use a volume to map a lockfile on my host machine and setup guard in the script.

```echo "heyheyhey"

if [ ! -f setup ]; then
  echo "Running first time setup..."
  touch setup
else
  echo "NOT running first time setup"
fi

sleep infinity
```

This time, with each restart, there is always one script that was run that didn't acknowledge the existence of that file.
<img width="297" alt="screen shot 2017-10-05 at 8 38 22 pm" src="https://user-images.githubusercontent.com/655011/31258501-8e15a944-aa0d-11e7-852c-d29fb2f7d601.png">

**Comment by ghsatpute** [comment]
@titangate 

This was interesting!

I dig up the code for docker compose, docker CLI, and tried some POC and as a conclusion, I can say **_your script is not running multiple times_**. 

This happens because the way Docker stores the logs for each container. These logs are stored in this directory `/var/lib/docker/containers/<container_id>/<container_id>-json.log`
(The ID is longer than what you'd see as an output of `docker ps` command. but you can match the first few digits)

Now, find the container ID by running `docker ps` command and `tailf` the logs. Try restarting the container, you'll see that the logs won't be repeated many times. 

If you run `docker logs <container_id>` command then you'll see the repeated logs, which will be processed by the `docker-compose`. 

So in short, it's just a problem with how the logs are processed. Your process is running only once.
